File: ios/Sources/AuthenticatorPlugin/AuthenticatorPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AuthenticatorPlugin)
public class AuthenticatorPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AuthenticatorPlugin"
    public let jsName = "Authenticator"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Authenticator()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/braxtron/capacitor/oauth/AuthenticatorPlugin.kt
package com.braxtron.capacitor.oauth

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "Authenticator")
class AuthenticatorPlugin : Plugin() {
    private val implementation = Authenticator()

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")

        val ret = JSObject()
        ret.put("value", implementation.echo(value))
        call.resolve(ret)
    }


    @PluginMethod
    fun hello(call: PluginCall) {
        val value = call.getString("value")

        val ret = JSObject()
        ret.put("value", implementation.echo("5"))
        call.resolve(ret)
    }
}
